internal/services/containers/kubernetes_cluster_node_pool_resource.go

```diff
@@ -1104,6 +1129,14 @@ func resourceKubernetesClusterNodePoolRead(d *pluginsdk.ResourceData, meta inter
 			d.Set("gpu_driver", string(pointer.From(v.Driver)))
 		}
 
+	gatewayPublicIPPrefixSize := 31
```

Comment:
why do we need to set `31` as the default value? Normally, we should set `gateway_public_ip_prefix_size` only when `props.GatewayProfile.PublicIPPrefixSize` exists